src/components/views/TableCellContextMenu.tsx

```diff
@@ -118,8 +148,16 @@ export const ContextMenu: React.FC<{
         <PlusIcon /> Insert {columns} to the right
       </ActionButton>
       <Separator />
+      <ActionButton
+        disabled={!headerDir}
+        onClick={() => headerDir && runCommand(addHeaderRow(headerDir))}
+      >
+        <PlusIcon /> Insert header row {headerDir || 'above'}
+      </ActionButton>
+      <Separator />
       <ActionButton onClick={() => runCommand(deleteRow)}>
-        <GrayDeleteIcon /> Delete {rows}
+        <GrayDeleteIcon /> Delete {headerDir === 'below' ? 'header ' : ''}
```

Comment:
Also this could be `Delete ${isInTableHeader(state, pos) ? 'header ' : ''}`

Reply:
updated